### DISCUSSION TITLE
integrating AdMob Adverts

### DISCUSSION THREAD

**Discussion by jeuxdaily** [airsdk/Adobe-Runtime-Support#3401] [General]
I need help to develop my project by integrating AdMob Adverts
I followed one of the codes but no result
thanks

**Comment by balaedit** [comment]
Are you using https://airnativeextensions.com/extension/com.distriqt.Adverts/ of **distriqt** ? if yes, @marchbold will sure help to solve your issue.

**Comment by marchbold** [comment]
What exactly is the issue? If you post your question in the Adverts repository we'll be notified of your question and be able to respond to you more quickly.

**Reply by jeuxdaily** [reply]
hello
thanks for your help
I had an old project that integrated AdMob ads, 
so bought extenstion distriqt.Adverts.
I actually found it difficult to integrate AdMob ads.

**Reply by jeuxdaily** [reply]
I want  to add a simple code Interstitial with this code
```as3
if (Adverts.isSupported)
{
	if (Adverts.service.isPlatformSupported( AdvertPlatform.PLATFORM_ADMOB ))
	{
		Adverts.service.setup( AdvertPlatform.PLATFORM_ADMOB );
		Adverts.service.addEventListener( AdvertsEvent.INITIALISED, initialisedHandler );
		Adverts.service.initialise();
			_interstitial = Adverts.service.interstitials.createInterstitialAd();					
		_interstitial.setAdUnitId( _interstitialUnitIdANDROID );
		_interstitial.addEventListener( InterstitialAdEvent.LOADED, interstitial_loadedHandler );
		_interstitial.addEventListener( InterstitialAdEvent.ERROR, interstitial_errorHandler );
		_interstitial.load( new AdRequestBuilder().build() );
	}
	else 
		{
			trace( "AdMob not supported" );
		}
}
```

but no result

APK not installed

**Reply by balaedit** [reply]
> I want to add a simple code Interstitial with this code `if (Adverts.isSupported) { if (Adverts.service.isPlatformSupported( AdvertPlatform.PLATFORM_ADMOB )) { Adverts.service.setup( AdvertPlatform.PLATFORM_ADMOB ); Adverts.service.addEventListener( AdvertsEvent.INITIALISED, initialisedHandler ); Adverts.service.initialise(); _interstitial = Adverts.service.interstitials.createInterstitialAd(); _interstitial.setAdUnitId( _interstitialUnitIdANDROID ); _interstitial.addEventListener( InterstitialAdEvent.LOADED, interstitial_loadedHandler ); _interstitial.addEventListener( InterstitialAdEvent.ERROR, interstitial_errorHandler ); _interstitial.load( new AdRequestBuilder().build() ); } else { trace( "AdMob not supported" ); } }`
> 
> but no result
> 
> APK not installed

APK not installed prompt means I thinks it's not extension's issue but it could be the architecture issue that bundle you are building.

**Reply by marchbold** [reply]
Can you elaborate on where your issue is, are you failing to produce an APK or is the installation failing?

**Comment by jeuxdaily** [comment]
hi
so
first :

I remove all code I just left extension
![1p](https://github.com/user-attachments/assets/2ecc93af-e802-4913-a1bf-ca60f45c3274)

![extension1](https://github.com/user-attachments/assets/cdab5382-feda-419f-977d-f14dfbcc4cb0)

the apk file is publising and is installing but not working

secode

I add this code 
	import com.distriqt.extension.adverts.AdSize;
	import com.distriqt.extension.adverts.AdView;
	import com.distriqt.extension.adverts.AdViewParams;
	import com.distriqt.extension.adverts.AdvertPlatform;
	import com.distriqt.extension.adverts.Adverts;
	import com.distriqt.extension.adverts.InterstitialAd;
	import com.distriqt.extension.adverts.builders.AdRequestBuilder;
	import com.distriqt.extension.adverts.builders.AdViewParamsBuilder;
	import com.distriqt.extension.adverts.events.AdViewEvent;
	import com.distriqt.extension.adverts.events.AdvertsEvent;
	import com.distriqt.extension.adverts.events.FullScreenContentEvent;
	import com.distriqt.extension.adverts.events.InterstitialAdEvent;

result Apk file not installing

it's only white this extension

**Reply by ajwfrost** [reply]
If you're using Animate .. are you just creating an APK file? ARMv7 format? Or do you have any configuration file settings for AAB or other formats?

It's worth trying to see what the ADB logcat output is saying during start-up...

**Reply by jeuxdaily** [reply]
thanks 
yes it's APK file ARMv8

I have another project it's work fine

but I get this error with this extension

**Reply by jeuxdaily** [reply]
![pp2](https://github.com/user-attachments/assets/f3b58ab3-0f16-45d2-9211-6f37b00a06a0)
this from adt
SDK 50.2.3.6

**Reply by ajwfrost** [reply]
Not sure why there's a failure in signing - there may be more information in the log file (or, just run AIR SDK Manager with the "Troubleshooting" page open).

But if it's warning about a lack of permissions, then things like adverts aren't likely to work as you should add the "INTERNET" permission in there to get network access...
```
<uses-permission android:name="android.permission.INTERNET"/>
```
within the `<android><manifestAdditions><manifest>` section of the app descriptor file.

**Reply by jeuxdaily** [reply]
thanks 
It's problem is only about distriqt .Adverts extension
but white another extension it's working good

https://docs.airnativeextensions.com/docs/adverts/platform/admob/

so how can I add extension in ADT command
thanks

**Comment by jeuxdaily** [comment]
I test some extension that it's working good but only extension distriqt.Adverts has a problem

**Comment by jeuxdaily** [comment]
this is a video

**Comment by jeuxdaily** [comment]
hi all
I solve my problem
All it's working
I'm working for Banner
thanks for your help
